spyder/plugins/maininterpreter.py

```diff
@@ -53,10 +54,11 @@ def __init__(self, parent, main):
             # the Python executable has already been set with pythonw.exe:
             self.set_option('executable',
                             executable.replace("pythonw.exe", "python.exe"))
+        self.set_custom_interpreters_list(executable, executable)

```

Comment:
I think this should be

```python
if not self.get_option('default'):
    self.set_custom_interpreters_list(executable, executable)
```

because we don't want the default interpreter to the list of custom ones.

Reply:
After trying this change I notice that we need at least one entry in the combobox. If not this triggers and error when loading the config (since there is no option to give to the current index in the combobox). The line where [this happens](https://github.com/spyder-ide/spyder/blob/aed5a5a190e92e8e653d92218582715039aa6f23/spyder/plugins/configdialog.py#L345)

Reply:
Mmmh, I don't like this because the default interpreter is going to be part of the list of custom ones too. Besides, users would find strange that there's one entry in the custom interpreters list if they haven't selected none.

Couldn't you find a way to make `configdialog` work with no interpreter?

Reply:
I could add a validation in `configdialog` on the line that I linked above to not raise and error, and also change a statement in `python_executable_changed` since currently if an invalid interpreter is found it will put the [default one as an option](https://github.com/spyder-ide/spyder/blob/7d786f889b42695636d5e81ad5642f4e19e8b2ad/spyder/plugins/maininterpreter.py#L153), and fallback to select the default interpreter option when the settings are applied with and empty custom interpreter probably.

Should I do this changes?